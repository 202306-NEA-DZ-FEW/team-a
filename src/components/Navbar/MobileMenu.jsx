import Link from "next/link";
import { useRouter } from "next/router";
import React, { useEffect, useState } from "react";
import { BiLogOut } from "react-icons/bi";
import { HiOutlineMenuAlt3, HiOutlineX } from "react-icons/hi";

function MobileMenu({ user, t, logOut }) {
  const [open, setopen] = useState(false);
  const [currentPath, setCurrentPath] = useState("/");
  const router = useRouter();
  useEffect(() => {
    // Get the current path
    const currentPath = router.asPath;
    setCurrentPath(currentPath);
  }, [router]);

  const handleClick = () => {
    setopen(false);
  };
  return (
    <nav className='lg:hidden no-scrollbar navbar-end z-50 no-scrollbar'>
      <button
        onClick={() => setopen(!open)}
        className='z-50 justify-center px-3 py-2 text-sm font-semibold btn-circle btn-ghost'
      >
        {open ? (
          <HiOutlineX className={`${open ? "text-white" : ""} w-6 h-6`} />
        ) : (
          <HiOutlineMenuAlt3
            className={`${open ? "text-white" : ""} w-6 h-6`}
          />
        )}
      </button>
      <div
        className={`${
          open ? "flex" : "hidden"
        } flex-col z-40 fixed justify-center w-screen h-screen bg-neutral top-0 left-0`}
      >
        <div className='w-full flex justify-center gap-4 items-center mt-10 '>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='en'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-gb rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='fr'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-fr rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='tr'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-tr rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='ar'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-dz rounded text-4xl'></span>
          </Link>
        </div>
        {user && (
          <Link
            onClick={handleClick}
            className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
            href='/'
          >
            {t("common:navbar:dashboard")}
          </Link>
        )}
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/'
        >
          {t("common:buttons:home")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/about'
        >
          {t("common:buttons:about")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/items'
        >
          {t("common:buttons:items")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/blogs'
        >
          {t("common:buttons:blogs")}
        </Link>
        {!user && (
          <Link
            href='/auth/sign-in'
            onClick={handleClick}
            className='btn mt-10 btn-active rounded-3xl normal-case font-normal tracking-wider w-[80%] self-center'
          >
            Login
          </Link>
        )}
        {user && (
          <button
            onClick={logOut}
            className='btn text-white text-xl normal-case font-light btn-ghost tracking-wider w-[80%] self-center'
          >
            <BiLogOut size={30} />
            Logout
          </button>
        )}
      </div>
    </nav>
  );
}

export default MobileMenu;
