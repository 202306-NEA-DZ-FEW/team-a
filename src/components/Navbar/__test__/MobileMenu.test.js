import renderer from "react-test-renderer";

import MobileMenu from "../MobileMenu";

it("it renders correctly", () => {
  const mockT = jest.fn();
  const tree = renderer
    .create(<MobileMenu currentPath='/' user={{}} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
