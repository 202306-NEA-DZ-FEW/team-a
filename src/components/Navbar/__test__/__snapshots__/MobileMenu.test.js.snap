// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<nav
  className="lg:hidden no-scrollbar navbar-end z-50 no-scrollbar"
>
  <button
    className="z-50 justify-center px-3 py-2 text-sm font-semibold btn-circle btn-ghost"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className=" w-6 h-6"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16M4 12h16m-7 6h7"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </button>
  <div
    className="hidden flex-col z-40 fixed justify-center w-screen h-screen bg-neutral top-0 left-0"
  >
    <div
      className="w-full flex justify-center gap-4 items-center mt-10 "
    >
      <a
        className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="fi fi-gb rounded text-4xl"
        />
      </a>
      <a
        className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="fi fi-fr rounded text-4xl"
        />
      </a>
      <a
        className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="fi fi-tr rounded text-4xl"
        />
      </a>
      <a
        className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="fi fi-dz rounded text-4xl"
        />
      </a>
    </div>
    <a
      className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
    <a
      className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
    <a
      className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
    <a
      className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
      href="/items"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
    <a
      className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
      href="/blogs"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
    <button
      className="btn text-white text-xl normal-case font-light btn-ghost tracking-wider w-[80%] self-center"
    >
      <svg
        fill="currentColor"
        height={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 13v-2H7V8l-5 4 5 4v-3z"
        />
        <path
          d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
        />
      </svg>
      Logout
    </button>
  </div>
</nav>
`;
